### DISCUSSION TITLE
Logging fails

### DISCUSSION THREAD

**Discussion by tsprasath** [Sunbird-Ed/Community#286] [Installation] [answered]
Hi Team,

CSP: GCP
I am trying to deploy a loggins it fails
Build version: release-5.1.0
Chart-name: fluent-bit

`TASK [logging : helm install] **************************************************
fatal: [localhost]: FAILED! => {
    "changed": true,
    "cmd": "helm upgrade --install --atomic --timeout 10m --force fluent-bit /var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Logging/kubernetes/helm_charts/logging/fluent-bit -n logging -f /tmp/fluent-bit.yaml",
    "delta": "0:00:01.670071",
    "end": "2022-11-29 15:07:13.073204",
    "rc": 1,
    "start": "2022-11-29 15:07:11.403133"
}

STDOUT:

Release "fluent-bit" does not exist. Installing it now.

STDERR:

Error: unable to build kubernetes objects from release manifest: [unable to recognize "": no matches for kind "ClusterRole" in version "rbac.authorization.k8s.io/v1beta1", unable to recognize "": no matches for kind "ClusterRoleBinding" in version "rbac.authorization.k8s.io/v1beta1"]

MSG:

non-zero return code
`

**Comment by beepdot** [comment, marked as answer]
What is the version of Kubernetes? If the `rbac.authorization.k8s.io/v1beta1` is removed in your Kubernetes version, please update the chart as per your Kubernetes API version.

**Reply by tsprasath** [reply]
@keshavprasadms  After updating facing following error while install elasticsearch

`TASK [logging : tempating variables] *******************************************
fatal: [localhost]: FAILED! => {
    "changed": false
}

MSG:

Could not find or access 'elasticsearch.yaml'
Searched in:
	/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Logging/kubernetes/ansible/roles/logging/templates/elasticsearch.yaml
	/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Logging/kubernetes/ansible/roles/logging/elasticsearch.yaml
	/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Logging/kubernetes/ansible/roles/logging/tasks/templates/elasticsearch.yaml
	/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Logging/kubernetes/ansible/roles/logging/tasks/elasticsearch.yaml
	/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Logging/kubernetes/ansible/templates/elasticsearch.yaml
	/var/lib/jenkins/workspace/Deploy/dev/Kubernetes/Logging/kubernetes/ansible/elasticsearch.yaml on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option
`

**Reply by beepdot** [reply]
Is it mentioned in the doc to install `elasticsearch`?

**Reply by tsprasath** [reply]
@keshavprasadms  nope elasticsearch not required.

**Reply by tsprasath** [reply]
@keshavprasadms  facing issue with oauth-proxy

`[2022/12/06 05:48:25] [main.go:75] WARNING: You are using alpha configuration. The structure in this configuration file may change without notice. You MUST remove conflicting options from your existing configuration.
[2022/12/06 05:48:25] [main.go:54] invalid configuration:
  provider missing setting: client-id
  missing setting: client-secret or client-secret-file`

**Reply by beepdot** [reply]
Have you set the required variables in your private repo like client secret and client id?
https://github.com/project-sunbird/sunbird-devops/blob/master/private_repo/ansible/inventory/dev/Core/secrets.yml#L127-L129

**Reply by tsprasath** [reply]
@keshavprasadms Variables values are blank.

**Reply by beepdot** [reply]
Please update them then.

**Reply by tsprasath** [reply]
Updated google OAuth it started to work.

**Comment by tsprasath** [comment]


**Reply by beepdot** [reply]
What is pointing to Azure?

**Reply by tsprasath** [reply]
@keshavprasadms  I assume `client-id missing setting: client-secret or client-secret-file` these variable are points to azure

**Reply by beepdot** [reply]
Its not related to Azure. Its google ouath client id and secret.
